frontend/src/app/modules/interventions/details/details.component.scss

```diff
@@ -1,41 +1,173 @@
 .mat-list-item {
-    .mat-icon {
-        margin-right: 10px;
-    }
+  .mat-icon {
+    margin-right: 10px;
+  }
 }
 .mat-raised-button {
-    .mat-icon {
-        font-size: 18px;
-        padding-left: 4px;
-        padding-top: 4px;
-    }
+  .mat-icon {
+    font-size: 18px;
+    padding-left: 4px;
+    padding-top: 4px;
+  }
 }
 
 .mat-card-content {
-    margin-top: 16px;
-    font-size: 16px;
+  margin-top: 16px;
+  font-size: 16px;
 
-    agm-map {
-        height: 200px;
-    }
+  agm-map {
+    height: 200px;
+  }
 
-    .map-container {
-        width: 60%;
-        margin: 20px auto;
-    }
+  .map-container {
+    width: 60%;
+    margin: 20px auto;
+  }
 }
 
 .mat-card-actions {
-    float: right;
+  float: right;
 }
 
 .embedded {
-    & .mat-card-actions {
-        padding: 0;
-    }
+  & .mat-card-actions {
+    padding: 0;
+  }
 }
 
 .title-wrapper {
-    display: inline-block;
+  display: inline-block;
 }
 
+.comment {
```

Comment:
Angular by defualt uses "emulated" view encapsulation per component basis, so you don't have to write nested selectors to prevent style leakage. See: https://angular.io/api/core/ViewEncapsulation

Think of it like css modules, e.g. if you write:
```
form {
  margin-top: 10px;
}
```
Angular will compile it to something like:
```
form[_ng12co93!s] {
  margin-top: 10px;
}
```
where generated class name is unique per every new component.

You could basically use simple element selectors in components css if you'd like:
```
p {
  // something
}

footer {
   // some other styles
}
``` 
these will not affect nor global scope, nor nested components unless you purposefully switch the component behaviour to use ViewEncapsulation.None.

Reply:
Also, if you really think you need to nest the selectors, I'd prefer to leverage the sass/scss nesting and "parent" selector so it is easier to read and understand what is nested in what.

e.g instead of: 
```
.component input {
   // styles
}

.component.some-class {
      // styles
}
```

I'd rather write:
```
.component {
   input {
     // styles
   }

   &.some-class {
     // styles
   }
}
```

First style IMO get's really messy when there is lot of nesting with lots of classes with similar names, while second style leverages spacing and you can fold nested styles in your IDE easily.

Please also bare in mind that lot's of styles in this app has been written during a hackathon so it is not probably the best reference for clean css practices, but I guess rule of thumb should be to keep as little nesting and as weak selectors as possible  ;)

Reply:
Thanks for that comment. I've simplified CSS by leveraging encapsulation.